### PR TITLE
refactor: look up default values on struct tag

### PR DESCRIPTION
**Pull request by fchikwekwe**
## Which problem is this PR solving?

- resolves part of #970 

## Short description of the changes

- add a struct tag lookup to getter funcs for fields that default to boolean true

### CODE REVIEW

File: config/mock.go [hunk 1]
```diff
@@ -423,11 +429,11 @@ func (m *MockConfig) GetIsDryRun() bool {
 	return m.DryRun
 }
 
-func (m *MockConfig) GetAddHostMetadataToTrace() bool {
+func (m *MockConfig) GetAddHostMetadataToTrace() (addHostMetadataToTrace bool, err error) {
 	m.Mux.RLock()
 	defer m.Mux.RUnlock()
 
-	return m.AddHostMetadataToTrace
+	return m.AddHostMetadataToTraceVal, m.AddHostMetadataToTraceErr
```
Comment (VinozzZ) [inline config/mock.go:436]: If we are changing `return m.AddHostMetadataToTrace` to use `m.AddHostMetadataToTraceVal` as the return value, we will need to assign `AddHostMetadataToTraceVal` to true in `newStartedApp` in `app_test.go`

File: config/mock.go [hunk 2]
```diff
@@ -75,13 +77,17 @@ type MockConfig struct {
 	DryRun                           bool
 	DryRunFieldName                  string
 	AddHostMetadataToTrace           bool
```
Comment (VinozzZ) [inline config/mock.go:79]: If we choose to use `AddHostMetadataToTraceVal`, we probably should delete `AddHostMetadataToTrace`.

Comment (fchikwekwe) [reply config/mock.go:79]: That's fair. I was trying to limit my changes a bit so I introduced some duplication here, but if I get the build passing I will try to not duplicate these values.

Comment (fchikwekwe) [reply config/mock.go]: Resolved this with the build fix. Thanks for the help!

**Review by VinozzZ — COMMENTED** [review]
Yahoo for fixing the default tag to true issue. I'm not certain that it's a good idea to parse the default tag each time when we try to read the value though. It feels wasted work for me. It would be nice if we could get the default value only once. I think that's maybe what you are working on in a following PR?

I'm thinking maybe it's worth taking the time to see whether using a pointer to represent the three states route works

**Comment by fchikwekwe** [comment]
@VinozzZ Yeah I just spoke with @kentquirk and he agreed that its maybe not the best idea to lookup struct tags for every getter func call. There is an option to cache the struct tag values to maybe make this better. But I think the consensus here is that using a `defaultTrue` type would be the best solution. I'll open a separate PR for that work.

**Pull request closed without merging**